TitleEdit/PluginServices/Lobby/LobbyService.Housing.cs

```diff
@@ -0,0 +1,156 @@
+﻿using System;
+using Dalamud;
+using Dalamud.Utility.Signatures;
+using FFXIVClientStructs.FFXIV.Client.Game;
+using FFXIVClientStructs.FFXIV.Client.Game.Object;
+using FFXIVClientStructs.FFXIV.Client.LayoutEngine;
+using FFXIVClientStructs.FFXIV.Client.System.Framework;
+using FFXIVClientStructs.Interop;
+using TitleEdit.Data.Persistence;
+using TitleEdit.Utility;
+
+namespace TitleEdit.PluginServices.Lobby;
+
+public unsafe partial class LobbyService
+{
+    const uint FurnitureOwnerId = 0x54455046;
+    
+    [Signature(" E8 ?? ?? ?? ?? FF C5 48 85 C0 0F 84")]
+    private readonly delegate*unmanaged<HousingFurniture*, HousingFurniture*, void*, int, HousingObject*> spawnFurnitureObject = null!;
+
+    [Signature("40 55 41 57 48 83 EC 58")]
+    private readonly delegate*unmanaged<HousingManager*, uint, void> initializeHousingLayout = null!;
+
+    [Signature("0F B7 0D ?? ?? ?? ?? 66 0F 45 C8 48 89 7C 24", ScanType = ScanType.StaticAddress)]
+    private readonly nint territoryTypeAddress;
+
+    [Signature("E8 ?? ?? ?? ?? 41 8B 4E 20")]
+    private readonly delegate*unmanaged<int, uint, ushort, int, byte, void> setInteriorFixture = null!;
+
+    private void InitializeHousingLayout(LocationModel model)
+    {
+        
+        var territory = model.Plots != null ? model.LayoutTerritoryTypeId : 649U;
+        SafeMemory.Write(territoryTypeAddress, territory);
```

Comment:
What's the purpose of this bit, I see that the game doesn't write the value in lobby and I assume the housing part needs it for some reason, but why is it always `649U`, can't it just be `model.LayoutTerritoryTypeId`?

Reply:
It needs to be a housing zone or the game will crash looking for how to handle furniture in the zone. I use `LayoutTerritoryTypeId` when its loading plots because it needs to know where to put the plots, but I put it into a random house for anywhere else. If the Layout territory is not a housing zone, it'll crash if switching from a layout with furniture.